soroban-env-host/src/test/poseidon/poseidon.rs

```diff
@@ -0,0 +1,288 @@
+                    if input1 != input2 {
+                        break;
+                    }
+                }
+
+                let perm1 = instance.permutation(&host, &input1).unwrap();
+                let perm2 = instance.permutation(&host, &input1).unwrap();
+                let perm3 = instance.permutation(&host, &input2).unwrap();
+                assert_eq!(perm1, perm2);
+                assert_ne!(perm1, perm3);
+            }
+        }
+    }
+
+    #[test]
+    fn kats() {
+        let host = Host::test_host();
+        let poseidon_2 = Poseidon::new((**POSEIDON_BLS_2_PARAMS).clone());
+        let input_2: Vec<Scalar> = vec![Scalar::from(0), Scalar::from(1)];
+        let perm_2 = poseidon_2.permutation(&host, &input_2).unwrap();
+        assert_eq!(
+            perm_2[0],
+            from_hex("0x1dc37ce34aeee058292bb73bff9acffce73a8a92f3d6d1daa8b77d9516b5c837")
+        );
+        assert_eq!(
+            perm_2[1],
+            from_hex("0x534cc8001b9c21da25d62749e136ea3d702651ba129f0d5ed7847cf81bc8b042")
+        );
+
+        let poseidon_3 = Poseidon::new((**POSEIDON_BLS_3_PARAMS).clone());
+        let input_3: Vec<Scalar> = vec![Scalar::from(0), Scalar::from(1), Scalar::from(2)];
+        let perm_3 = poseidon_3.permutation(&host, &input_3).unwrap();
+        assert_eq!(
+            perm_3[0],
+            from_hex("0x200e6982ac00df8fa65cef1fde9f21373fdbbfd98f2df1eb5fa04f3302ab0397")
+        );
+        assert_eq!(
+            perm_3[1],
+            from_hex("0x2233c9a40d91c1f643b700f836a1ac231c3f3a8d438ad1609355e1b7317a47e5")
+        );
+        assert_eq!(
+            perm_3[2],
+            from_hex("0x2eae6736db3c086ad29938869dedbf969dd9804a58aa228ec467b7d5a08dc765")
+        );
+    }
+
+    #[test]
+    fn hadeshash_x5_255_3_test_cases() {
+        let host = Host::test_host();
+        // poseidonperm_x5_255_3
+        let poseidon = Poseidon::new((**POSEIDONPERM_X5_255_3).clone());
+
+        // Input
+        let input: Vec<Scalar> = vec![
+            from_hex("0x0000000000000000000000000000000000000000000000000000000000000000"),
+            from_hex("0x0000000000000000000000000000000000000000000000000000000000000001"),
+            from_hex("0x0000000000000000000000000000000000000000000000000000000000000002"),
+        ];
+
+        // Expected output
+        let expected_output: Vec<Scalar> = vec![
+            from_hex("0x28ce19420fc246a05553ad1e8c98f5c9d67166be2c18e9e4cb4b4e317dd2a78a"),
+            from_hex("0x51f3e312c95343a896cfd8945ea82ba956c1118ce9b9859b6ea56637b4b1ddc4"),
+            from_hex("0x3b2b69139b235626a0bfb56c9527ae66a7bf486ad8c11c14d1da0c69bbe0f79a"),
+        ];
+
+        // Run the permutation
+        let result = poseidon.permutation(&host, &input).unwrap();
+
+        // Verify the output matches expected values
+        assert_eq!(result.len(), expected_output.len());
+        for (i, (actual, expected)) in result.iter().zip(expected_output.iter()).enumerate() {
```

Comment:
Since it seems like Debug is implemented for the `Scalar`, couldn't this just be `assert_eq` on the vectors? Vector assertions are pretty readable IMO, and the test would be simplified a lot (especially given that you repeat this particular snippet a lot)

Reply:
done